src/infi/clickhouse_orm/database.py

```diff
@@ -79,6 +81,56 @@ def gen():
                 yield ('\n'.join(batch) + '\n').encode('utf-8')
         self._send(gen())
 
+    def insert_try_best(self, model_instances, batch_size = 1000, is_create = False, insert_count = 0, first_recu = True):
+        cache = []
+        table_name = None
+        for elem in model_instances:
+            if is_create == False:
+                self.create_table(elem)
+                is_create = True
+                table_name = elem.table_name()
+            cache.append(elem)
+            if len(cache) >= batch_size:
+                try:
+                    # Must ensure that batch_size > = bulksize
+                    self.insert(cache, batch_size=batch_size)
+                    insert_count += len(cache)
+                    cache = []
+                except:
+                    # binary insertion
+                    if len(cache) > 1:
+                        self.insert(self, cache[:(len(cache) / 2)],
```

Comment:
my mistake, as you said should be call insert_try_best recursively, 
this func to ensure insert when a bulk insert faild because of some dirty data.